staging/src/k8s.io/apiextensions-apiserver/pkg/cmd/server/options/options.go

```diff
@@ -92,6 +95,10 @@ func (o CustomResourceDefinitionsServerOptions) Validate() error {
 
 // Complete fills in missing options.
 func (o *CustomResourceDefinitionsServerOptions) Complete() error {
+	// set feature gate emulation version before Validate().
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.EmulationVersion) {
+		return utilfeature.DefaultMutableVersionedFeatureGate.SetEmulationVersion(serverversion.Effective.EmulationVersion())
```

Comment:
I am surprised the apiextensions-apiserver needed to set this separately

Reply:
This is a global. But it is also a different server. The other `SetEmulationVersion` will not be called when starting this server.

Reply:
ack, didnt realize the context of this was `CustomResourceDefinitionsServerOptions` which is the standalone path. usually our usage is for kube-aggregator, apiextexensions, and apiserver to all share a single process.  

Is there a way to keep all this logic in one place so that the feature gate emulation version can be set when `serverversion` emulation version is set? Is sthere a dependency issue preventing feature gate from using serverversion package?